### PR TITLE
8 add new comment

### PR DESCRIPTION
**Pull request by moshkh**

### CODE REVIEW

File: src/Components/AddComments.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      author: currUser,
+      body: newComment,
+      created_at: date.toISOString(),
+      votes: 0,
+    };
+    setCommentId((currCommentId) => {
+      return currCommentId - 1;
+    });
+
+    setComments((currComments) => {
```
Comment (callumu21) [inline src/Components/AddComments.jsx]: We shouldn't be optimistically rendering a comment here. As comment deletion and editing will depend on a specific ID, we need to make sure that ID exists in the database

File: src/Components/AddComments.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      setErr({ msg: "Oops something went wrong - try posting again!" });
+    });
+    setButtonText("Comment Posted! Post Again?");
+    setNewComment("");
+    setErr({ msg: "" });
+  };
+
+  return (
+    <form className="add-comment" onSubmit={handleCommentSubmit}>
+      <h4>{err.msg}</h4>
```
Comment (callumu21) [inline src/Components/AddComments.jsx]: For better user experience and UI, I'd recommend moving this error message to below the button. That's where they'll click for the error to happen, so their eyes are already there 👀

File: src/Components/AddComments.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    };
+    setCommentId((currCommentId) => {
+      return currCommentId - 1;
+    });
+
+    setComments((currComments) => {
+      return [commentSubmitted, ...currComments];
+    });
+
+    postCommentToArticle(articleId, currUser, newComment).catch((err) => {
```
Comment (callumu21) [inline src/Components/AddComments.jsx]: You'll need a way of making sure this request only happens once, even if a user presses the button multiple times on a slow connection where the initial post request hasn't been completed

**Review by callumu21 — CHANGES_REQUESTED** [review]
Nice start, Musab! Just a couple of changes need to be made before this can be approved:

1. We shouldn't be optimistically rendering the added comment in this case, as functionality for editing and deleting a comment will depend on the correct ID.
2. You need a way of stopping a user from pressing the post button multiple times before the post request has been completed.

File: src/Components/AddComments.jsx (hunk 2 above)
Comment (moshkh) [reply src/Components/AddComments.jsx]: Change made

File: src/Components/AddComments.jsx (hunk 1 above)
Comment (moshkh) [reply src/Components/AddComments.jsx]: Change made

File: src/Components/AddComments.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        setCommentAdded(true);
+        setDisableButton(false);
+        setButtonText("Comment Posted! Post Again?");
+      })
+      .catch((err) => {
+        setDisableButton(false);
+        setButtonText("Post Comment");
+        setErr({ msg: "Oops something went wrong - try posting again!" });
+      });
+    setNewComment("");
```
Comment (callumu21) [inline src/Components/AddComments.jsx]: I'd move resetting the comment box into the .then block. That means that if the post is unsuccessful, the user won't lose their entire comment and they can try again. Similarly, resetting the error message should only happen on a successful request.

**Review by callumu21 — APPROVED** [review]
Very nice! I've left a comment with one small change to be made, but you should be good to merge after that 👍

File: src/Components/AddComments.jsx (hunk 4 above)
Comment (moshkh) [reply src/Components/AddComments.jsx]: change made

**Pull request merged**